### PR TITLE
fix: add catch handling for already exists

### PR DESCRIPTION
**Pull request by sutterj**
# Pull Request

## Proposed Changes

- Fixes an issue caused by removing the try/catch around the exists call

### Author/Contributor

- [ ] If documentation is needed for this change, has that been included in this pull request
- [X] run `npm run lint` and fix any linting issues that have been introduced
- [X] run `npm run test` and run tests
- [ ] If publishing new data to the public (scorecards, security scan results, code quality results, live dashboards, etc.), please request review from `@jeffrey-luszcz`

### Reviewer

- [X] Label as either `bug`, `documentation`, `enhancement`, `infrastructure`, `maintenance`, or `breaking`

### CODE REVIEW

**Comment by sutterj** [comment]
The alternative to this is letting removing the exists check altogether and letting the error be thrown when the repo fails to create, but doing this exists call first saves a few api calls.

File: src/server/repos/controller.ts [hunk 1]
```diff
@@ -49,16 +49,35 @@ export const createMirrorHandler = async ({
...
+        }
+      })
+      .catch((error) => {
+        // catch and rethrow the error if the repo already exists
+        if ((error as Error).message.includes('already exists')) {
+          throw error
+        }
+
+        // if there is a real error, then we log it and throw it
+        if (!(error as Error).message.includes('Not Found')) {
```
Comment (ajhenry) [inline src/server/repos/controller.ts:76]: We don’t have to do this check since we throw above, so we can just throw here instead

**Review by ajhenry — APPROVED** [review]
Looks good other than the duplicated check 👍

File: src/server/repos/controller.ts (hunk 1 above)
Comment (ajhenry) [reply src/server/repos/controller.ts:76]: :doh: I’m wrong here and this logic is right, carry on!

**Pull request merged**